app/components/connection_banner/connection_banner.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {
+    Text,
+    View,
+} from 'react-native';
+import Animated, {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import CompassIcon from '@components/compass_icon';
+import {ANNOUNCEMENT_BAR_HEIGHT} from '@constants/view';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type Props = {
+    isDisconnected: boolean;
+}
+
+const getStyle = makeStyleSheetFromTheme((theme: Theme) => {
+    const bannerContainer = {
+        flex: 1,
+        paddingHorizontal: 10,
+        overflow: 'hidden',
+        flexDirection: 'row',
+        alignItems: 'center',
+        marginHorizontal: 8,
+        borderRadius: 7,
+    };
+    return {
+        background: {
+            backgroundColor: theme.sidebarBg,
+        },
+        bannerContainerNotConnected: {
+            ...bannerContainer,
+            backgroundColor: theme.centerChannelColor,
+        },
+        bannerContainerConnected: {
+            ...bannerContainer,
+            backgroundColor: theme.onlineIndicator,
+        },
+        wrapper: {
+            flexDirection: 'row',
+            flex: 1,
+            overflow: 'hidden',
+        },
+        bannerTextContainer: {
+            flex: 1,
+            flexGrow: 1,
+            marginRight: 5,
+            textAlign: 'center',
+            color: theme.centerChannelBg,
+        },
+        bannerText: {
+            ...typography('Body', 100, 'SemiBold'),
+        },
+    };
+});
+
+const ConnectionBanner = ({
+    isDisconnected,
+}: Props) => {
+    const intl = useIntl();
+    const closeTimeout = useRef<NodeJS.Timeout | null>();
+    const height = useSharedValue(0);
+    const theme = useTheme();
+    const [visible, setVisible] = useState(false);
```

Comment:
perhaps use `useAppState` so that it only becomes visible when the app is in the foreground as the WS will get disconnected when the app is in the background.

Reply:
I wonder what is the expected behaviour here. The websocket disconnect, and is disconnected until it connects again. So... should we have an optimistic approach when we get into the app (be either from the background or from a cold start)? But if the websocket cannot reconnect... the banner should show eventually, right? When is eventually?

@matthewbirtch in case you have given some thought to this already.

Reply:
That is how is done in v1